### PR TITLE
Async params

### PR DESCRIPTION
**Pull request by VX792**
First of the PRs related to asynchronous controllers and components.
This doesn't do anything other than reading the is_async parameters and storing their values as a member.

By default, both parameters are optional and their default value is false.

### CODE REVIEW

**Review by bijoua29 — APPROVED** [review]

File: hardware_interface/src/component_parser.cpp [hunk 1]
```diff
@@ -211,6 +213,20 @@ std::string parse_data_type_attribute(const tinyxml2::XMLElement * elem)
...
+ * Parses an XMLElement and returns the value of the is_async attribute.
+ * Defaults to "false" if not specified.
+ *
+ * \param[in] elem XMLElement that has the data_type attribute.
+ * \return boolean specifying the if the value read was true or false.
+ */
+bool parse_is_async_attribute(const tinyxml2::XMLElement * elem)
+{
+  const tinyxml2::XMLAttribute * attr = elem->FindAttribute(kIsAsyncAttribute);
+  return attr ? strcmp(attr->Value(), "true") == 0 : false;
```
Comment (bmagyar) [inline hardware_interface/src/component_parser.cpp:227]: a little fool-proofing: I think it'd be worth also checking for `True` for all those who grew up with Python

**Review by bmagyar — DISMISSED** [review]

File: hardware_interface/src/component_parser.cpp (hunk 1 above)
Comment (VX792) [reply hardware_interface/src/component_parser.cpp]: I've decided to use strcasecmp instead.

Comment (bmagyar) [reply hardware_interface/src/component_parser.cpp]: Nice, thanks!

**Review by destogl — COMMENTED** [review]
Thanks!

**Pull request merged**